mne/decoding/time_gen.py

```diff
@@ -320,14 +379,18 @@ def score(self, epochs=None, y=None):
                 raise RuntimeError('Please predict() epochs first or pass '
                                    'epochs to score()')
 
-        # clean gat.score() attributes
-        for att in ['scores_', 'scorer_', 'y_true_']:
-            if hasattr(self, att):
-                delattr(self, att)
-
         # Check scorer
-        # XXX Need API to identify proper scorer from the clf
-        self.scorer_ = accuracy_score if self.scorer is None else self.scorer
+        if self.scorer is not None:
+            self.scorer_ = self.scorer
+        else:
+            if is_classifier(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = accuracy_score
+            elif is_regressor(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = mean_squared_error
+            else:
+                raise ValueError('Could not find a scoring metrics for '
+                                 '`clf=%s` and `predict_method=%s`. Manually'
+                                 ' define scorer. ' % (self.clf, self.scorer))
```

Comment:
Ah I see. Sorry, what I said is irrelevant, from what I can tell, you're right and it should be `self.predict_method`.